### ISSUE TITLE
Return all exit angles for roundabouts

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#4178]
When a navigation UI wants to render icons for maneuvers, it can use the `intersections` object to render a graphical representation of the upcoming intersection (including valid lane information).

However, roundabouts are problematic.  They're not represented by a single intersection object, we get an intersection object for each roundabout exit that the route uses.  We supply no data for exits that the route does not pass, making rendering of roundabout icons incomplete.

We should try to figure out a place to put additional roundabout metadata.

One option would be to add an `intersection` object to the `StepManeuver` for the roundabout:

```
                     "maneuver" : {
                        "exit" : 2,
                        "bearing_after" : 63,
                        "type" : "roundabout",
                        "modifier" : "straight",
                        "location" : [
                           -77.029316,
                           38.909145
                        ],
                        "bearing_before" : 43,
                        "intersection" :   // <- New intersection object when type=roundabout
                        {
                           "location" : [
                              -77.029316,
                              38.909145
                           ],
                           "entry" : [
                              false,   // Don't use the first exit
                              true,    // The second exit is the one to take
                              false    // Third exit is not on our route at all, but we include it for rendering
                           ],
                           "in" : 1,
                           "out" : 0,
                           "bearings" : [
                              60,
                              225,
                              240
                           ]
                        },
                     },
```

We have precedence for this with the `exit` property.

**Comment by MoKob** [comment]
@danpat I like the idea for rendering all roundabout exits. I don't know whether your proposal  would be considered a breaking change though, since we would change the interpretation of an intersection.

E.g. in your example, taking the third exit would look like entering a roundabout to the left instead of to the right. Also, the entry flags are quite confusing to me, since what I would read from your example is that I would exit the same way I enter (in == 1 == true). But that is most likely me being not fully awake yet :).

How would you feel about adding a full shape to the object? So we wouldn't change the meaning of the intersection itself but add something like you describe as a property?

```
{
  "roundabout_shape": {
	"entry": [false,true,false,true],
	"exit" : [true,false,true,true],
	"bearings": [0,90,180,270]
  }
}
```

This shape would describe the roundabout in full:
 - four roads
 - bearings 0, 180, and 270 allow exiting the roundabout
 - bearings 90, 270 allow entering it (making all roads but 270 oneways)
 - bearings show how to draw arrows
 - based on the maneuver the correct bearings can be highlighted (or we add in/out in our processing)

We could do this as an additive change, without breaking the meaning of the current intersections.

**Comment by danpat** [comment]
Yes, a dedicated `roundabout_shape` object probably makes more sense than re-using the existing `intersection` object.

The end goal is to have enough information to render something like this:

![screen shot 2017-06-24 at 7 25 42 am](https://user-images.githubusercontent.com/1892250/27509306-6187d31c-58ae-11e7-8882-6e84c22ef989.png)

Many roundabouts have sliproads for entry/exit, so I don't think we want to confuse the rendering by showing *all* connected intersections - I think the exits are probably sufficient.  On roundabouts, it's usually very clear that you shouldn't do a >90 degree turn to exit the roundabout.

**Comment by emiltin** [comment]
as a side node, turbo roundabouts require that you choose the correct lane when entering: https://en.wikipedia.org/wiki/Roundabout#Turbo_roundabouts
